Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -734,6 +734,7 @@ extension CarPlayManager: CPMapTemplateDelegate {
             passiveLocationProvider.locationManager.resumeTripSession()
         }
         delegate?.carPlayManagerDidEndNavigation(self)
```

Comment:
@mapbox/navigation-ios, since this method is deprecated we still have to call it. At the same time it causes a warning while building in Xcode. Workaround, similar to this one doesn't work for protocols:
```swift
class TestClass {

    @available(*, deprecated, renamed: "newMethod()")
    func deprecatedMethod() {
        print("Deprecated method")
    }

    func newMethod() {
        print("New method")
    }
}

protocol TestClassDeprecations {

    func deprecatedMethod()
}

extension TestClass: TestClassDeprecations {

}

// 'deprecatedMethod()' is deprecated: renamed to 'newMethod()'
TestClass().deprecatedMethod()

// Not deprecation warning.
(TestClass() as TestClassDeprecations).deprecatedMethod()
```

At this point I do not have any ideas other than making this delegate method `obsolete` and stop calling it.

Reply:
#1967 looks similar to what you tried above, but with a protocol. It might’ve worked because the protocol was an Objective-C protocol.

Reply:
I've tried a couple more ways, but none of them worked. At this point I'm more inclined to remove calls to deprecated delegate method to prevent warnings while building for CocoaPods.